roles/openshift_health_checker/openshift_checks/docker_storage.py

```diff
@@ -0,0 +1,181 @@
+            vals["data_total_bytes"] + vals["vg_free_bytes"])
+        vals["metadata_pct_used"] = 100 * vals["metadata_used_bytes"] / (
+            vals["metadata_total_bytes"] + vals["vg_free_bytes"])
+        msg = ""
+        if vals["data_threshold"] < vals["data_pct_used"]:
+            vals["failed"] = True
+            msg = (
+                "Docker thinpool data usage percentage {:.1f} "
+                "is higher than threshold {:.1f}."
+            ).format(vals["data_pct_used"], vals["data_threshold"])
+        if vals["metadata_threshold"] < vals["metadata_pct_used"]:
+            vals["failed"] = True
+            msg = msg + "\n" if msg else ""
+            msg += (
+                "\nDocker thinpool metadata usage percentage {:.1f} "
+                "is higher than threshold {:.1f}."
+            ).format(vals["metadata_pct_used"], vals["metadata_threshold"])
+
+        vals["msg"] = msg or "Thinpool usage is below thresholds."
+        return vals
+
+    def _get_vg_free(self, pool, task_vars):
+        # pool looks like "foo--vg-docker--pool"; vg and lv joined by hyphen, inner hyphens doubled!
+        match = re.match(r'((?:[^-]|--)+)-(?!-)', pool)  # matches up to the first single hyphen
+        if not match:
+            raise OpenShiftCheckException(
+                "Host's docker says it is using LVM storage pool with invalid name '{}'".format(pool)
+            )
+        vg_name = match.groups()[0].replace("--", "-")
+        vgs_cmd = "/sbin/vgs --noheadings -o vg_free --select vg_name=" + vg_name
+        # should return free space like "  12.00g" if the VG exists; empty if it does not
+
+        ret = self.execute_module("command", {"_raw_params": vgs_cmd}, task_vars)
+        if ret.get("failed") or ret.get("rc", 0) != 0:
+            raise OpenShiftCheckException(
+                "Is LVM installed? Failed to run /sbin/vgs "
+                "to determine docker storage usage:\n" + ret.get("msg", "")
+            )
+        size = ret.get("stdout", "").strip()
+        if not size:
+            raise OpenShiftCheckException(
+                "Host's docker says it is using LVM storage pool '{pool}'\n"
+                "which we expect to come from local VG '{vg}'.\n"
+                "However, /sbin/vgs did not find this VG. Is docker for this host"
+                "running and using the storage on the host?".format(pool=pool, vg=vg_name)
+            )
+        return size
+
+    @staticmethod
+    def _convert_to_bytes(string):
+        units = dict(
+            k=1024,
+            m=1024**2,
+            g=1024**3,
+            t=1024**4,
+            p=1024**5,
+        )
+        string = string or ""
+        match = re.match(r'(\d+(?:\.\d+)?)\s*(\w)?', string)  # float followed by optional unit
+        if not match:
+            raise ValueError("Cannot convert to a byte size: " + string)
+        number, unit = match.groups()
+        return float(number) * units.get(str(unit).lower(), 1)
```

Comment:
If the unit is anything that is not in the `units` dict we may incorrectly assume it is bytes.
I'd rather check explicitly. Empty = bytes, one of the units in the dicts = multiplication factor, otherwise error.